drivers/dac/ad5758/ad5758.c

```diff
@@ -112,17 +116,19 @@ static int32_t ad5758_spi_reg_read(struct ad5758_dev *dev,
 
 	ret = spi_write_and_read(dev->spi_desc, buf, 4);
 	if (ret < 0)
-		return FAILURE;
+		goto error;
 
 	if ((dev->crc_en) &&
-	    (ad5758_compute_crc8(buf, 3) != buf[3])) {
-		printf("%s: CRC Error.\n", __func__);
-		return FAILURE;
-	}
+	    (ad5758_compute_crc8(buf, 3) != buf[3]))
+		goto error;
 
 	*reg_data = (buf[1] << 8) | buf[2];
 
 	return SUCCESS;
+
+error:
+	pr_err("%s: CRC Error.\n", __func__);
```

Comment:
It's not necessarily a CRC error. This message should only show in case ad5758_compute_crc8() returns error. Right now it shows even at spi_write_and_read() errors.